jdaviz/configs/cubeviz/plugins/tools.py

```diff
@@ -91,15 +91,40 @@ class SpectrumPerSpaxel(ProfileFromCube):
     action_text = 'See spectrum at a single spaxel'
     tool_tip = 'Click on the viewer and see the spectrum at that spaxel in the spectrum viewer'
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._is_moving = False
+
+    def activate(self):
+        super().activate()
+        for k in ("y_min", "y_max"):
+            self._profile_viewer.state.add_callback(k, self.on_limits_change)
+
+    def deactivate(self):
+        for k in ("y_min", "y_max"):
+            self._profile_viewer.state.remove_callback(k, self.on_limits_change)
+        super().deactivate()
+
+    def on_limits_change(self, *args):
+        if not self._is_moving:
+            self._previous_bounds = self._profile_viewer.get_limits()
+
     def on_mouse_move(self, data):
+        self._is_moving = False
```

Comment:
could this just be in the mouseleave if-statement (just to avoid toggling it False > True on every event)?

Reply:
OK, sure. Done!